internal/btf/btf.go

```diff
@@ -307,8 +312,7 @@ type Handle struct {
 
 // NewHandle loads BTF into the kernel.
 //
-// Returns an error if BTF is not supported, which can
-// be checked by IsNotSupported.
+// Returns ErrNotSupported.
```

Comment:
This sounds like `NewHandle` returns `ErrNotSupported` unconditionally. Maybe change this to `Returns ErrNotSupported if BTF is not supported.`